avm/res/insights/data-collection-rule/main.bicep

```diff
@@ -97,19 +75,30 @@ resource avmTelemetry 'Microsoft.Resources/deployments@2024-03-01' = if (enableT
   }
 }
 
-resource dataCollectionRule 'Microsoft.Insights/dataCollectionRules@2021-09-01-preview' = {
-  kind: kind
+resource dataCollectionRule 'Microsoft.Insights/dataCollectionRules@2023-03-11' = {
+  kind: dataCollectionRuleProperties.kind
   location: location
   name: name
   tags: tags
-  properties: {
-    dataSources: dataSources
-    destinations: destinations
-    dataFlows: dataFlows
-    dataCollectionEndpointId: dataCollectionEndpointId
-    streamDeclarations: streamDeclarations
-    description: description
-  }
+  properties: union(
+    dataCollectionRuleProperties.kind == 'Linux' || dataCollectionRuleProperties.kind == 'Windows'
+      ? {
+          dataSources: dataCollectionRuleProperties.dataSources
+          dataFlows: dataCollectionRuleProperties.dataFlows
+          destinations: dataCollectionRuleProperties.destinations
+          dataCollectionEndpointId: dataCollectionRuleProperties.?dataCollectionEndpointId
```

Comment:
Should be 
`dataCollectionRuleProperties.?dataCollectionEndpointResourceId`

Will not create the comment as a suggestion as this requires updates in other places too.